client/apps/edit/components/content2/sections/header/index.jsx

```diff
@@ -0,0 +1,162 @@
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      progress: null
+    }
+  }
+
+  onChange = (key, value) => {
+    this.props.article.set(key, value)
+  }
+
+  onChangeHero = (key, value) => {
+    const heroSection = this.props.article.heroSection
+    heroSection.set(key, value)
+    this.onChange('hero_section', heroSection.attributes)
+  }
+
+  onChangeLeadParagraph = (html) => {
+    this.onChange('lead_paragraph', html)
+  }
+
+  onUpload = (image, width, height) => {
+    this.onChangeHero('url', image)
+  }
+
+  onProgress = (progress) => {
+    this.setState({ progress })
+  }
+
+  onRemoveImage = () => {
+    this.onChangeHero('url', '')
+  }
+
+  renderTitle(article) {
+    return (
+      <PlainText
+        content={article.attributes.title}
+        onChange={this.onChange}
+        name='title'
+        placeholder='Title' />
+    )
+  }
+
+  renderFeatureDeck(article) {
+    return (
+      <PlainText
+        content={article.heroSection.get('deck')}
+        onChange={this.onChangeHero}
+        name='deck'
+        placeholder='Deck (optional)' />
+    )
+  }
+
+  renderFileUpload(prompt) {
+    return (
+      <FileInput
+        type='simple'
+        onUpload={this.onUpload}
+        prompt={prompt}
+        video={true}
+        onProgress={this.onProgress} />
+    )
+  }
+
+  renderImage(article) {
+    const isFullscreen = article.heroSection.get('type') === 'fullscreen'
+    const hasUrl = article.heroSection.get('url').length
+    const prompt = isFullscreen ? 'Add Background' : 'Add Image or Video'
+
+    if (isFullscreen && hasUrl) {
+      return (
+        <div className='edit-header__image-container has-image'>
+          {this.renderFileUpload('Change Background')}
+        </div>
+      )
+    } else if(hasUrl) {
+      return (
+        <div
+          className='edit-header__remove'
+          onClick={this.onRemoveImage}>
+          <IconRemove />
+        </div>
+      )
+    } else {
+      return (
+        <div className='edit-header__image-container'>
+          {this.renderFileUpload(prompt)}
+          {this.state.progress && this.renderProgress()}
+        </div>
+      )
+    }
+  }
+
+  renderProgress() {
+    return (
+      <div className='upload-progress-container'>
+        <div
+          className='upload-progress'
+          style={{width: (this.state.progress * 100) + '%'}}>
+        </div>
+      </div>
+    )
+  }
+
+  renderLeadParagraph(article) {
+    return (
+      <Paragraph
+        html={article.get('lead_paragraph')}
+        onChange={this.onChangeLeadParagraph}
+        placeholder='Lead Paragraph (optional)'
+        type='lead_paragraph'
+        linked={false}
+        stripLinebreaks={true}
+        layout={article.get('layout')} />
+    )
+  }
+
+  render() {
+    const { article } = this.props
+    const isFeature = article.get('layout') === 'feature'
+    const isClassic = article.get('layout') === 'classic'
```

Comment:
Did we do any research on Editorial Standard articles that have `hero_section` that are `image` or `video` types? I can't remember how we said we'll handle those.

Reply:
This is on my todo list this week -- was planning to keep close to the original hero_section implementation

Reply:
Was also considering adding an optional limit to the image_collection to max out at one image, which would allow us to kill the image component